### DISCUSSION TITLE
bgfill on settextcolor not present when using a gradient background

### DISCUSSION THREAD

**Discussion by erikkll** [Bodmer/TFT_eSPI#2310] [Q&A - General] [answered]
When using setTextColor, boolean bgfill can be used to overwrite text. However, when using a gradient background, only setTextColor with a single color can be used. There is no boolean for bgfill. 

I also tried text padding as per the examples, but this doesnt work for smooth fonts, i think?

**Comment by Bodmer** [comment]
Have a look at the ReadMe news item 7, that may address the problem.

If not provide an example sketch that demonstrates the issue.

**Comment by Bodmer** [comment, marked as answer]
I realised that you might be expecting the library to redraw the gradient. The library does not have that capability so the sketch would have to clear the correct area. A better approach is to use a sprite and redraw the background and text in the sprite for each update and then push to the screen.

**Comment by erikkll** [comment]
I was indeed thinking that, like the background, the library might be able to dynamically redraw the gradient. I'll give drawing to a sprite and then pushing it for each update a try! Thanks for the suggestion.

**Comment by erikkll** [comment]
Okay i tried using a sprite and it works perfectly! Thanks for the suggestion.
